File: longest_subarray_of_1s_after_deleting_one_element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        best = 0
        zero = 0
        left = 0
        
        for right in range(n):
            if nums[right] == 0:
                zero += 1
            
                
            while zero > 1:
                if nums[left] == 0:
                    zero -= 1
                left += 1
                
            best = max(best, right - left) #should actually be right - left + 1 - 1 to include proper bounds
        return best

File: longestSubarrayOf1sAfterDeletingOneElement.go
func longestSubarray(nums []int) int {
	zero := 0
	left := 0
	best := 0

	for i := 0; i < nums.length; i++ {
		if nums[i] == 0 {
			zero += 1
		}

		for zero > 1 {
			if nums[left] == 0 {
				zero -= 1
			}
			left += 1
		}
		best = max(best, i-left+1-1)
	}
	return best
}